File: 백준/Bronze/11653. 소인수분해/소인수분해.py
import sys
import math

def prime_factors(n):
    factors = []
    
    # 2부터 시작해서 나누기
    while n % 2 == 0:
        factors.append(2)
        n //= 2
    
    # 3 이상의 홀수로 나누기
    for i in range(3, int(math.sqrt(n)) + 1, 2):
        while n % i == 0:
            factors.append(i)
            n //= i
    
    # 남은 소수 처리
    if n > 2:
        factors.append(n)
    
    return factors

# 입력 받기
N = int(sys.stdin.readline().strip())

# N이 1인 경우 아무것도 출력하지 않음
if N != 1:
    factors = prime_factors(N)
    for factor in factors:
        print(factor)

File: 백준/Bronze/11653. 소인수분해/소인수분해.js
const fs = require('fs');

// 입력 받기
const N = parseInt(fs.readFileSync('/dev/stdin').toString().trim(), 10);

function primeFactors(n) {
    const factors = [];
    let i = 2;
    
    while (i * i <= n) { // i가 n의 제곱근까지 반복
        if (n % i === 0) { // i로 나누어 떨어지면
            while (n % i === 0) { // 계속 나누어떨어질 때까지
                factors.push(i);
                n /= i;
            }
        }
        i++;
    }
    
    if (n > 1) { // n이 1보다 크면, n 자체가 소수
        factors.push(n);
    }
    
    return factors;
}

// N이 1인 경우 아무것도 출력하지 않음
if (N !== 1) {
    const factors = primeFactors(N);
    factors.forEach(factor => console.log(factor));
}
